src/main/java/com/ee/employeesportal/controller/EmployeeController.java

```diff
@@ -13,36 +14,25 @@
 @RequiredArgsConstructor
 @RequestMapping("/api/employees")
 public class EmployeeController {
-    private final EmployeeRepository employeeRepository;
+//    private final EmployeeRepository employeeRepository;
 
+    private final EmployeeService employeeService;
     @GetMapping("")
     public List<Employee> getAllEmployees(){
-    return employeeRepository.getAllEmployees();
+//    return employeeRepository.getAllEmployees();
+       return employeeService.getAllEmployees();
     }
 
-    @GetMapping("/{empId}")
-    public ResponseEntity<Employee> getEmployeeById(@PathVariable(name = "empId") Long empId){
-        Employee employeeById = employeeRepository.getEmployeeById(empId);
-        if(employeeById != null)
-            return ResponseEntity.ok(employeeById);
-
-        return ResponseEntity.badRequest().build();
+    @GetMapping("/sort/firstName")
+    public List<Employee> getAllEmployeesSortByName(){
+        return employeeService.getAllEmployeesSortByName();
     }
-
-    @PostMapping("")
-    public ResponseEntity<Employee> createEmployee(@RequestBody Employee employee){
-        employee = employeeRepository.createEmployee(employee);
-        return ResponseEntity.status(HttpStatus.CREATED).body(employee);
+    @GetMapping("/sort/dateOfJoin")
+    public List<Employee> getAllEmployeesSortByDateOfJoin(){
+        return employeeService.getAllEmployeesSortByDateOfJoin();
     }
-
-    @PutMapping("/{empId}")
-    public Employee updateEmployeeById(@PathVariable(name = "empId") Long empId,@RequestBody Employee employee) {
-        employee.setEmpId(empId);
-        return employeeRepository.updateEmployeeById(employee);
-    }
-
-    @DeleteMapping("/{empId}")
-    public Employee deleteEmployeeById(@PathVariable(name = "empId") Long empId) {
-        return employeeRepository.deleteEmployeeById(empId);
+    @PutMapping("/")
```

Comment:
For Update APIs we accept `id` as part of URL like /api/employees/{id} and use that id instead of what is coming in `employee` object.